homework-submission/reactweek1/src/App.js

```diff
@@ -0,0 +1,123 @@
+    this.deadLineHandler = this.deadLineHandler.bind(this);
+    this.removeHandler = this.removeHandler.bind(this);
+    this.addHandler = this.addHandler.bind(this);
+    this.state = {
+      list: [
+        {
+          "id": 1,
+          "description": "Get out of bed",
+          "deadline": "2017-09-11",
+          "done": true
+        },
+        {
+          "id": 2,
+          "description": "Brush teeth",
+          "deadline": "2017-09-10",
+          "done": false
+        },
+        {
+          "id": 3,
+          "description": "Eat breakfast",
+          "deadline": "2017-09-09",
+          "done": false
+        },
+      ],
+      newDescription: '',
+      newDeadline: ''
+    }
+  }
+
+  changeToDone = (event) => {
+    const id = event.target.id;
+    const listCopy = this.state.list.slice(0);
+    const itemIndex = listCopy.findIndex(el => { return el.id === Number(id) });
+    listCopy[itemIndex].done = !listCopy[itemIndex].done;
+    this.setState({
+      list: listCopy
+    });
+  }
+  descriptionHandler = (newDescription) => {
+    const state = this.state;
+    state['newDescription'] = newDescription;
+    this.setState(state)
+  };
+
+  deadLineHandler = (newDeadLine) => {
+    const state = this.state;
+    state['newDeadline'] = newDeadLine;
+    this.setState(state)
+  };
+
+  addHandler = () => {
+    const state = this.state;
+    const newItem = {
+      'id': this.state.list.length + 1,
+      'description': this.state.newDescription,
+      'deadline': this.state.newDeadline,
+      'done': false
+    };
+    this.state.list.push(newItem);
+    // this.state.newDeadline = '';
+    // this.state.newDeadline = ''; commented out these two cus it faces some problem through the test but dont know what the better way is
+    this.setState(state)
+  };
+
+  removeHandler = (i) => {
+    const state = this.state;
+    this.state.list.splice(i, 1);
+    this.setState(state);
+  };
+
+  itemHandler() {
+    if (this.state.list.length === 0) return <p>No items... </p>;
+
+    return <div>
+      <ul>
+        {
+          this.state.list.map((el, i) => {
+            return <div key={i}>
+              <li className='todoList'>
+                <button onClick={() => this.removeHandler(i)}>Remove</button>
+                <input id={el.id} type="checkbox" onChange={this.changeToDone} defaultChecked={el.done} />
+                <span style={{ textDecoration: el.done ? "line-through" : null }}>
+                  {el.id} {el.description} - {el.deadline}
+                </span>
+              </li>
+            </div>;
+          })
+        }
+      </ul>
+      <label>
+        Enter description :
+              <input type="text" onChange={(e) => this.descriptionHandler(e.target.value)} value={this.state.list.newDescription} />
+      </label>
+      <label>
+        Enter deadline :
+              <input type="text" onChange={(e) => this.deadLineHandler(e.target.value)} value={this.state.list.newDeadline} />
```

Comment:
There is a major problem here. The user can insert whatever they want here :) How can we make sure that they only insert a date formatted how we want it?
hint: [take a look at the `date` input](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/input/date)

Reply:
wow, fency stuff  :) was a good one thanks.